dioptas/widgets/plot_widgets/ColormapDialog.py

```diff
@@ -0,0 +1,133 @@
+        closeButton.clicked.connect(self.accept)
+        closeButton.setAutoDefault(False)
+        layout.addRow(buttonBox)
+
+    def _gradientComboBoxCurrentIndexChanged(self, index: int):
+        if index < 0:
+            return
+        gradient = self._gradientComboBox.itemData(index, QtCore.Qt.UserRole)
+        self.sigCurrentGradientChanged.emit(gradient)
+
+    def setCurrentGradient(self, gradient: dict):
+        """Set the currently selected gradient
+
+        If the gradient is not available, a 'Custom' item is added for it.
+        """
+        for name, description in pyqtgraph.graphicsItems.GradientEditorItem.Gradients.items():
+            if gradient['mode'] == description['mode'] and gradient['ticks'] == description['ticks']:
+                self._gradientComboBox.setCurrentText(name.capitalize())
+                return
+
+        icon = self._createQIconFromGradient(gradient)
+        # Block signals to avoid emitting with previously selected gradient since index changes
+        wasBlocked = self._gradientComboBox.blockSignals(True)
+        self._gradientComboBox.insertItem(0, icon, 'Custom', gradient)
+        self._gradientComboBox.blockSignals(wasBlocked)
+        self._gradientComboBox.setCurrentIndex(0)
+
+    def getCurrentGradient(self) -> dict:
+        """Returns the currently selected gradient"""
+        return self._gradientComboBox.currentData()
+
+    def _rangeChanged(self):
+        minimum, maximum = self.getRange()
+        if maximum < minimum:
+            self.setRange(maximum, minimum)
+            return
+        self.sigRangeChanged.emit(minimum, maximum)
+
+    def setRange(self, minimum: float, maximum: float):
+        """Set the data range (min, max) of the colormap"""
+        if maximum < minimum:
+            minimum, maximum = maximum, minimum
+        if (minimum, maximum) == self.getRange():
+            return
+        self._minEdit.setText(self._minEdit.validator().locale().toString(float(minimum)))
+        self._maxEdit.setText(self._maxEdit.validator().locale().toString(float(maximum)))
+        self._rangeChanged()
+
+    def getRange(self) -> tuple[float, float]:
+        """Returns the data range of the colormap (min, max)"""
+        minimum, validated = self._minEdit.validator().locale().toDouble(self._minEdit.text())
+        if not validated:
+            minimum = 1 
+        maximum, validated = self._maxEdit.validator().locale().toDouble(self._maxEdit.text())
+        if not validated:
+            maximum = minimum
+        return minimum, maximum
+
+    @staticmethod
+    def _createQIconFromGradient(gradient: dict) -> QtGui.QIcon:
+        """Generates a QIcon from a pyqtgraph gradient"""
+        gradientEditorItem = pyqtgraph.graphicsItems.GradientEditorItem.GradientEditorItem()
+        gradientEditorItem.setLength(100)
+        gradientEditorItem.restoreState(gradient)
+        qgradient = gradientEditorItem.getGradient()
```

Comment:
Using the `GradientEditorItem` to convert the pyqtgraph `gradient` into a `QLinearGradient` since there does not seems to exist a function to do this.